### PR TITLE
Home work automation #5

### PR DESCRIPTION
**Pull request by TanyaB777**

### CODE REVIEW

**Comment by AntBelyi** [comment]
Please rename the PR title. Probably something like "Home work automation #5" ???

File: src/main/java/hw_at_17/HomePageTests.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    @Test(priority = 1)
+    public void verifyProductCategoryShowing() {
+
+        final Wait<WebDriver> wait = new WebDriverWait(getDriver(), Duration.ofSeconds(5)).withMessage("Element was not found");
+
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector("[class='css-cbwxzx']")));
+
+        WebElement gardenItem = getDriver().findElement(By.cssSelector("[class='css-cbwxzx']"));
+        gardenItem.click();
```
Comment (AntBelyi) [inline src/main/java/hw_at_17/HomePageTests.java]: Test is always failed

File: src/main/java/hw_at_17/HomePageTests.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+        assertTrue(searchResultText.isDisplayed());
+
+        List<WebElement> searchResult = getDriver().findElements(By.cssSelector("[class='css-1wxaaza']"));
+
+        for (WebElement element : searchResult) {
+            Assert.assertNotNull(element);
+        }
+
+        assertFalse(searchResult.isEmpty());
```
Comment (AntBelyi) [inline src/main/java/hw_at_17/HomePageTests.java]: On the row 43 you verify that all elements not null, but then on the row 46 you verify again that list is not empty. Why do you need this extra verification.
I would recommend to verify something like:
for (WebElement element : searchResult) {
            Assert.assertTrue(element.isDisplayed());
}

But this verification assertFalse(searchResult.isEmpty()); can be romoved.
How do you think?

File: src/main/java/hw_at_17/SearchResultPageTests.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+        WebElement searchInput = getDriver().findElement(By.id("search"));
+        searchInput.sendKeys(wordToSearch, Keys.ENTER);
+
+        WebElement searchResultText = getDriver().findElement(By.cssSelector("[data-testid='total-count']"));
+
+        List<WebElement> searchResult = getDriver().findElements(By.cssSelector("[class='css-1wxaaza']"));
+
+        for (WebElement element : searchResult) {
+            //System.out.println(element.getText());
```
Comment (AntBelyi) [inline src/main/java/hw_at_17/SearchResultPageTests.java]: Do we need this line?

File: src/main/java/hw_at_17/SearchResultPageTests.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    public void verifyEmptySearchInput() {
+
+        final String SEARCH_PRODUCT_TEXT = " ";
+
+        WebElement searchInput = getDriver().findElement(By.id("search"));
+        searchInput.sendKeys(SEARCH_PRODUCT_TEXT, Keys.ENTER);
+
+        WebElement searchResultText = getDriver().findElement(By.cssSelector("[data-testid='total-count']"));
+
+        System.out.println(searchResultText.getText());
```
Comment (AntBelyi) [inline src/main/java/hw_at_17/SearchResultPageTests.java]: please remove this  System.out.println

File: run.byclasses.xml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        <classes>
+            <class name = "hw_at_17.SearchResultPageTests" />
+            <class name = "hw_at_17.SearchResultPageTests" >
+                <methods>
+                    <exclude name = "verifyPositiveSearchInput"/>
+                </methods>
+            </class>
+        </classes>
+    </test>
+</suite>
```
Comment (AntBelyi) [inline run.byclasses.xml:12]: You need to create 3 separate .xml files
image

File: src/main/java/hw_at_17/HomePageTests.java (hunk 2 above)
Comment (TanyaB777) [reply src/main/java/hw_at_17/HomePageTests.java]: fixed

File: src/main/java/hw_at_17/SearchResultPageTests.java (hunk 3 above)
Comment (TanyaB777) [reply src/main/java/hw_at_17/SearchResultPageTests.java]: fixed

File: src/main/java/hw_at_17/SearchResultPageTests.java (hunk 4 above)
Comment (TanyaB777) [reply src/main/java/hw_at_17/SearchResultPageTests.java]: fixed

File: run.byclasses.xml (hunk 5 above)
Comment (TanyaB777) [reply run.byclasses.xml:12]: fixed

File: run.negative_tests.xml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<?xml version = "1.0" encoding = "UTF-8"?>
+<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
+
+<suite name="Nagative Tests" verbose="1">
+    <test name="Negative Tests">
+        <classes>
+            <class name = "aqa_hw_3.SearchResultPageTests" />
```
Comment (AntBelyi) [inline run.negative_tests.xml]: do you really need this row?

**Comment by TanyaB777** [comment]
> Please rename the PR title. Probably something like "Home work automation #5" ???

fixed

File: run.negative_tests.xml (hunk 6 above)
Comment (TanyaB777) [reply run.negative_tests.xml]: fixed

**Pull request merged**